### PR TITLE
Feature/741/demo k3s

### LINKED ISSUE #643: [BUG] When installing the Jenkins role with k3s it shows an error
When installing the Jenkins role in k3s using 

```
- include_role:
    name: jenkins
```

image

image

### LINKED ISSUE #739: [FEATURE] Install specific k3s version
Currently k3s installed in the latest version.

As per best practices, all software packages shall install a specific version to ensure idempotency.

### LINKED ISSUE #741: [FEATURE] Move all standard demo use cases to k3s
At the moment, standard demo use cases are on microk8s. We shall move them to k3s and plan also on sunsetting microk8s support.

### PR DESCRIPTION
**Pull request by kristofre**
Closes #741 
Closes #739
Closes #643 

Tested with the following use cases:
```
demo_all - which includes:


- include_role:
    name: demo-release-validation-srg-gitlab

- include_role:
    name: demo-ar-workflows-gitlab

- include_role:
    name: demo-monaco-gitops

- include_role:
    name: demo-ar-workflows-ansible

```

### CODE REVIEW

File: user-skel/ansible_collections/ace_box/ace_box/roles/app-simplenode/files/repos/simplenode/helm/simplenodeservice/templates/ingress.yaml [hunk 1]
```diff
@@ -11,13 +11,12 @@
...
-
+{{ if eq .Values.ingress.mode "regular"}}
 ---
 apiVersion: networking.k8s.io/v1
 kind: Ingress
 metadata:
   annotations:
-    kubernetes.io/ingress.class: "{{ .Values.ingress.class }}"
     nginx.ingress.kubernetes.io/canary: "{{ if .Values.ingress.isCanary }}true{{ else }}false{{ end }}"
     nginx.ingress.kubernetes.io/canary-weight: "{{ .Values.ingress.canaryWeight }}"
```
Comment (eemrdog) [inline user-skel/ansible_collections/ace_box/ace_box/roles/app-simplenode/files/repos/simplenode/helm/simplenodeservice/templates/ingress.yaml:21]: is "canaryWeight" redundant here? As I see, if the ingress.mode is "canary" this weight parameter is not used. weight is only 1 or 0

File: user-skel/ansible_collections/ace_box/ace_box/roles/app-simplenode/files/repos/simplenode/helm/simplenodeservice/templates/ingress.yaml [hunk 2]
```diff
@@ -40,3 +40,38 @@ spec:
...
+---
+apiVersion: traefik.containo.us/v1alpha1
+kind: TraefikService
+metadata:
+  name: {{ .Chart.Name }}
+spec:
+  weighted:
+    services:
+      - name: {{ .Chart.Name }}-0
+        weight: 1
```
Comment (eemrdog) [inline user-skel/ansible_collections/ace_box/ace_box/roles/app-simplenode/files/repos/simplenode/helm/simplenodeservice/templates/ingress.yaml:69]: can canaryWeight value be used here?

**Review by eemrdog — APPROVED** [review]
just a comment on the canaryweight parameter of app-simplenode role when using ingress.mode canary

File: user-skel/ansible_collections/ace_box/ace_box/roles/app-simplenode/files/repos/simplenode/helm/simplenodeservice/templates/ingress.yaml (hunk 2 above)
Comment (kristofre) [reply user-skel/ansible_collections/ace_box/ace_box/roles/app-simplenode/files/repos/simplenode/helm/simplenodeservice/templates/ingress.yaml:69]: For the `nginx` ingress controller that is possible, but not for `traefik`, as it requires a weighting for each target service. I wanted to keep backwards compatibility for those who are using `microk8s` in combination with `nginx` ingress.

**Pull request merged**